docs/docs/user/getting-started/components.md

```diff
@@ -0,0 +1,529 @@
+
+## Producer
+
+_Producer_ inherits from [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Publishes (write) events to a Kafka cluster.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      # Holds configuration to use as values for the streams bootstrap produce helm
+      # chart.
+      # More documentation on ProducersApp:
+      # https://github.com/bakdata/streams-bootstrap
+      - type: producer # required
+        name: producer # required
+        namespace: namespace # required
+        # `app` can contain application-specific settings, hence  the user is free to
+        # add the key-value pairs they need. The fields that are specific to producers
+        # are marked with `# producer-specific`
+        app: # required
+          streams: # required, producer-specific
+            brokers: ${broker} # required
+            schemaRegistryUrl: ${schema_registry_url}
+            outputTopic: output_topic
+            extraOutputTopics: 
+              output_role1: output_topic1
+              output_role2: output_topic2
+          nameOverride: override-with-this-name # kafka-app-specific
+          kkey1:
+            key4: value3
+          key2: value1
+          key3: value2
+        # Topic(s) from which the component will read input
+        from:
+          topics: # required
+            ${pipeline_name}-input-topic:
+              type: input # required
+              role: topic-role
+        # Topic(s) into which the component will write output
+        to:
+          topics: # required
+            ${pipeline_name}-output-topic:
+              type: output # required
+              role: topic-role
+              keySchema: key-schema
+              valueSchema: value-schema
+              partitions_count: 1
+              replication_factor: 1
+              configs: 
+                config1: config1
+          models: 
+            model: model
+        # Pipeline prefix that will prefix every component name. If you wish to not 
+        # have any prefix you can specify an empty string.
+        prefix: ${pipeline_name}-
+        # Helm repository configuration, the default value is given here
+        repoConfig:
+          repositoryName: bakdata-streams-bootstrap # required
+          url: https://bakdata.github.io/streams-bootstrap/ # required
+          repo_auth_flags:
+            username: user
+            password: pass
+            ca_file: /home/user/path/to/ca-file
+            insecure_skip_tls_verify: false
+        version: "2.7.0"
+    ```
+
+
+### deploy
+
+A Kubernetes job or cron job is deployed.
+The output topics are created, and schemas are registered in the Schema registry if configured.
+
+### destory
+
+The associated Kubernetes resources are removed.
+
+### reset
+
+Producers are not affected by reset as they are stateless.
+
+### clean
+
+The output topics of the Kafka producer are deleted as well as all associated schemas in the Schema Registry.
+
+## KafkaSinkConnector
```

Comment:
There is also the KafkaConnector super class

Reply:
Added fake links in both connectors to a page that would go over KafkaConnector, noted that the 2 connectors here are subclasses of it.